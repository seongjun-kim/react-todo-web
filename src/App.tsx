import React from "react";
import logo from "./logo.svg";
import Styled, { keyframes } from "styled-components";

const spin = keyframes`
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
`;

const Container = Styled.div`
  text-align: center;
`;

const Header = Styled.div`
  background-color: #282c34 !important;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
`;

const Image = Styled.img`
  height: 40vmin;
  pointer-events: none;
  @media (prefers-reduced-motion: no-preference) {
    // styled-components에서는 해당 컴포넌트가 직접 애니메이션을 
    // 수행하므로 클래스명을 추가로 명시하지 않아도 된다!
    // .App-logo {
      animation: ${spin} infinite 20s linear;
    // }
  }
`;

const AppLink = Styled.a`
  color: #61dafb
`;

function App() {
  return (
    <Container>
      <Header>
        <Image src={logo} alt="logo" />
        <p>
          Edit <code>src/App.js</code> and save to reload.
        </p>
        <AppLink
          href="https://reactjs.org"
          target="_blank"
          rel="noopener noreferrer"
        >
          Learn React
        </AppLink>
      </Header>
    </Container>
  );
}

export default App;
