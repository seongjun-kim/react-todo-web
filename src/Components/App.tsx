import React from 'react';
import Styled from 'styled-components';
import { Button, Input, ToDoItem } from 'Components';

const Container = Styled.div`
  min-height: 100vh;
  display: flexx;
  align-items: center;
  justify-content: center;
  flex-direction: column;
`;

const Contents = Styled.div`
  display:flex;
  background-color: #fff;
  flex-direction: column;
  padding: 20px;
  border-radius: 8px;
  box-shadow: 5px 5px 10px rgba(0,0,0,0.2);
`;

const InputContainer = Styled.div`
  display: flex;
  align-self: center;
  width: 105%;
  justify-content: space-around;
`;

function App() {
  return (
    <Container>
      <Contents>
        <ToDoItem label="todo" onDelete={() => alert('Delete!')} />
        <InputContainer>
          <Input placeholder="Enter what to do..." onChange={(text) => console.log(text)} />
          <Button
            label="Add"
            onClick={() => {
              alert('Add!');
            }}
          />
        </InputContainer>
      </Contents>
    </Container>
  );
}

export default App;
