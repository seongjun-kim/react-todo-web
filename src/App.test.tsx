import React from 'react';
import { render, screen } from '@testing-library/react';
import App from './App';

describe('<App />', () => {
  it('renders component correctly', () => {
    // const { container } = render(<App />);
    // const linkElement = screen.getByText(/learn react/i);
    // expect(linkElement).toBeInTheDocument();
    // // expect(container.getElementsByTagName('a')).toHaveLength(1);
    // // expect(container.getElementsByTagName('a')[0]).toHaveTextContent(
    // //   'Learn React'
    // // );
    // const appLogo = screen.getByAltText('logo');
    // expect(appLogo).toBeInTheDocument();
    // expect(appLogo).toHaveAttribute('src', 'logo.svg');
    // expect(container.getElementsByTagName('p')).toHaveLength(1);
    // expect(container.getElementsByTagName('p')[0]).toHaveTextContent(
    //   'Edit src/App.js and save to reload.',
    // );
    // expect(container).toMatchSnapshot();
  });
});
