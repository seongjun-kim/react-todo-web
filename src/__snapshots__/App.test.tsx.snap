// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders component correctly 1`] = `
<div>
  <div
    class="sc-bdvvtL bKhKNO"
  >
    <div
      class="sc-gsDKAQ ipDMAC"
    >
      <img
        alt="logo"
        class="sc-dkPtRN gBvcKg"
        src="logo.svg"
      />
      <p>
        Edit 
        <code>
          src/App.js
        </code>
         and save to reload.
      </p>
      <a
        class="sc-hKwDye boCHbV"
        href="https://reactjs.org"
        rel="noopener noreferrer"
        target="_blank"
      >
        Learn React
      </a>
    </div>
  </div>
</div>
`;
